resource/subfunctions/kerberos_realm.md

```diff
@@ -0,0 +1,274 @@
+
+&nbsp;
+
+{: .warning }
+> _Alter operations in pyracf require READ access to `IRR.IRRSMO00.PRECHECK` in the `XFACILIT` class_
+> _This function will not produce output unless the user running the command has this access._
+
+&nbsp;
+
+Alter an existing **Kerberos Realm** profile in the **REALM** class.
+
+#### 📥 Parameters
+* `kerberos_realm_name`<br>
+  The name of the **Kerberos Realm** profile being defined to the **REALM** class.
+
+* `traits`<br>
+  A dictionary of **traits/attributes** that should be given to the resource on creation. See [Traits](../../base/segments_traits_operators#traits) to see what all of the valid **Resource Traits** are.
+
+#### 📤 Returns
+* `Union[dict, bytes]`<br>
+  Returns a **Security Result dictionary** or **Security Request XML bytes** if the `ResourceAdmin.generate_requests_only` class attribute is set to `True`.
+
+#### ❌ Raises
+* `SecurityRequestError`<br>
+  Raises `SecurityRequestError` when the **Return Code** of a **Security Result** returned by IRRSMO00 is **NOT** equal to `0`.
+* `AlterOperationError`<br>
+  Raises `AlterOperationError` when the **z/OS userid** supplied cannot be altered because it does **NOT** exist.
+* `SegmentTraitError`<br>
+  Raises `SegmentTraitError` when the dictionary of **traits/attributes** provided contains one or more **unknown** traits.
+
+#### 💻 Example
+
+###### Python REPL
+```python
+from pyracf import ResourceAdmin
+resource_admin = ResourceAdmin()
+resource_admin.alter_kerberos_realm("TSTREALM", traits={"kerb:encryption_algorithms": "AES128"})
+{'securityResult': {'resource': {'name': 'TSTREALM', 'class': 'REALM', 'operation': 'set', 'requestId': 'ResourceRequest', 'info': ['Definition exists. Add command skipped due  to precheck option'], 'commands': [{'safReturnCode': 0, 'returnCode': 0, 'reasonCode': 0, 'image': 'RALTER  REALM             (TSTREALM)  KERB     (ENCRYPT     (AES128))', 'messages': ['ICH11009I RACLISTED PROFILES FOR REALM WILL NOT REFLECT THE UPDATE(S) UNTIL A SETROPTS REFRESH IS ISSUED.']}]}, 'returnCode': 0, 'reasonCode': 0}}
+```
+
+###### Security Result Dictionary as JSON
+```json
+{
+  "securityResult":{
+    "resource":{
+      "name":"TSTREALM",
+      "class":"REALM",
+      "operation":"set",
+      "requestId":"ResourceRequest",
+      "info":[
+        "Definition exists. Add command skipped due  to precheck option"
+      ],
+      "commands":[
+        {
+          "safReturnCode":0,
+          "returnCode":0,
+          "reasonCode":0,
+          "image":"RALTER  REALM             (TSTREALM)  KERB     (ENCRYPT     (AES128))",
+          "messages":[
+            "ICH11009I RACLISTED PROFILES FOR REALM WILL NOT REFLECT THE UPDATE(S) UNTIL A SETROPTS REFRESH IS ISSUED."
+          ]
+        }
+      ]
+    },
+    "returnCode":0,
+    "reasonCode":0
+  }
+}
+```
+
+## `ResourceAdmin.extract_kerberos_realm()`
+
+```python
+  def extract_kerberos_realm( self, kerberos_realm_name: str ) -> Union[dict, bytes]:
+```

```

Comment:
```python
def extract_kerberos_realm(self, kerberos_realm_name: str) -> Union[dict, bytes]:
```

Reply:
Remove leading spaces.

Reply:
Leading spaces and unnecessary trailing spaces and indents removed across all function definitions.

Reply:
change `(self, kerberos_realm_name: str )` to `(self, kerberos_realm_name: str)`

Reply:
Sorry, somehow kerb got missed. now fixed in extract and delete.